pykrige/uk.py

```diff
@@ -281,13 +297,26 @@ def __init__(self, x, y, z, variogram_model='linear', variogram_parameters=None,
                 self.variogram_function = variogram_function
         else:
             self.variogram_function = self.variogram_dict[self.variogram_model]
+
         if self.verbose:
             print("Initializing variogram model...")
+
+        # see comment in ok.py about 'use_psill' kwarg...
+        if variogram_parameters is None:
+            vp_temp = None
+        elif use_psill:
+            vp_temp = variogram_parameters
+        else:
+            if self.variogram_model in ['gaussian', 'spherical', 'exponential', 'hole-effect']:
+                vp_temp = [variogram_parameters[0] - variogram_parameters[2],
+                           variogram_parameters[1], variogram_parameters[2]]

```

Comment:
Still don't understand this. If it is indeed needed, maybe we could put this chunk of the code in a separate private function in `core.py`?